src/Compiler/NodeVisitor/TransformForeachWithIteratorNodeVisitor.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Efabrica\PHPStanLatte\Compiler\NodeVisitor;
+
+use Efabrica\PHPStanLatte\Compiler\NodeVisitor\Behavior\ActualClassNodeVisitorBehavior;
+use PhpParser\Node;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\New_;
+use PhpParser\Node\Stmt\Expression;
+use PhpParser\Node\Stmt\Foreach_;
+use PhpParser\NodeVisitorAbstract;
+
+/**
+ * changed output from:
+ * <code>
+ * foreach ($iterator = $ʟ_it = new LR\CachingIterator($stringList, $ʟ_it ?? null) as $string)
+ * </code>
+ *
+ * to:
+ * <code>
+ * $iterator = $ʟ_it = new LR\CachingIterator($stringList, $ʟ_it ?? null)
+ * foreach ($stringList as $string) {
+ * </code>
+ */
+final class TransformForeachWithIteratorNodeVisitor extends NodeVisitorAbstract implements PostCompileNodeVisitorInterface
+{
+    use ActualClassNodeVisitorBehavior;
+
+    /**
+     * @return Node[]|null
+     */
+    public function leaveNode(Node $node): ?array
+    {
+        if (!$node instanceof Foreach_) {
+            return null;
+        }
+
+        if (!$node->expr instanceof Assign) {
+            return null;
+        }
+
+        if (!$node->expr->expr instanceof Assign) {
+            return null;
+        }
+
+        if (!$node->expr->expr->expr instanceof New_) {
```

Comment:
Maybe use more specific checks here to avoid false matches. At least check that zhis creates new instance of CachingIterator class.

Reply:
Well it's not that simple without scope :)

Reply:
I would simply check just that `New_` node `class` property is `Name` node with `Latte\Essential\CachingIterator` No neeed for anything more advanced

Reply:
Ok thats doable